sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -114,15 +116,15 @@ private void initializeCryptoClients() {
         if (localKeyCryptographyClient != null) {
             return;
         }
-        if (key.getKeyType().equals(RSA) || key.getKeyType().equals(RSA_HSM)) {
-            localKeyCryptographyClient = new RsaKeyCryptographyClient(key, cryptographyServiceClient);
-        } else if (key.getKeyType().equals(EC) || key.getKeyType().equals(EC_HSM)) {
-            localKeyCryptographyClient = new EcKeyCryptographyClient(key, cryptographyServiceClient);
-        } else if (key.getKeyType().equals(OCT)) {
-            localKeyCryptographyClient = new SymmetricKeyCryptographyClient(key, cryptographyServiceClient);
+        if (key.get().getKeyType().equals(RSA) || key.get().getKeyType().equals(RSA_HSM)) {
+            localKeyCryptographyClient = new RsaKeyCryptographyClient(key.get(), cryptographyServiceClient);
+        } else if (key.get().getKeyType().equals(EC) || key.get().getKeyType().equals(EC_HSM)) {
+            localKeyCryptographyClient = new EcKeyCryptographyClient(key.get(), cryptographyServiceClient);
+        } else if (key.get().getKeyType().equals(OCT)) {
+            localKeyCryptographyClient = new SymmetricKeyCryptographyClient(key.get(), cryptographyServiceClient);
         } else {
             throw logger.logExceptionAsError(new IllegalArgumentException(String.format(
-                "The Json Web Key Type: %s is not supported.", key.getKeyType().toString())));
+                "The Json Web Key Type: %s is not supported.", key.get().getKeyType().toString())));
```

Comment:
This would be more readable with a `switch` statement

Reply:
Expandable String enum, cannot do that, as discussed.